presto-main/src/main/java/com/facebook/presto/util/LongValueCompressor.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.facebook.presto.util;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+public class LongValueCompressor
```

Comment:
> I can in-line this. It was even my first approach, but I had a bug there. Writing tests for an operator in isolation is not an easy task to do. So this is why I decided to extract this.

Fair enough, but, in that case, it would suffice to have a specialized version that knows just how to assemble an id from a stageId, taskId and rowId. My main concern is with the over-abstraction it brings -- i.e., a generic utility for packing bits in a long that's used in exactly one way and one place. It's more code to understand, maintain, etc.

> Regarding performance, IMO it is premature-optimization, such simple abstraction should be easily optimized by HotSpot.

Possibly, and it may not even matter in practice given everything else that goes on in a query execution. However, from what I see in the code, it's unlikely it will perform as well as the straightforward version.